temba/msgs/models.py

```diff
@@ -1396,10 +1429,23 @@ def evaluate_template(cls, text, context, org=None, url_encode=False, partial_va
         # returns tuple of output and errors
         return evaluate_template(text, context, url_encode, partial_vars)
 
+    @classmethod
+    def get_outgoing_metadata(cls, quick_replies, expressions_context, contact, org, channel):
+        for counter, reply in enumerate(quick_replies):
+            (text, errors) = Msg.evaluate_template(reply, expressions_context, org=org)
+            if text:
+                if contact.is_test:
+                    quick_replies[counter] = text
+                else:
+                    channel_type = Channel.get_type_from_code(channel.channel_type)
+                    quick_replies[counter] = text[:channel_type.quick_reply_text_size]
```

Comment:
This should happen in each channel handler - which means this method is probably unnecessary. Just call Msg.evaluate_template on each quick_reply in Msg.create_outgoing